src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs

```diff
@@ -114,8 +114,10 @@ private List<Slice> ExtractIdsFromQuery(IndexQueryServerSide query)
                 if (documentId.Equals(EqualPrefix))
                 {
                     var id = q.SubSegment(EqualPrefix.Length);
+                    var idAsString = id.ToString();
```

Comment:
Everything is done on StringSegment, we should not materialize it to string if possible